### PR TITLE
feat: ✨ add syn-divider

### LINKED ISSUE #270: feat: ✨  Create syn-divider
# Checklist

- [x] I have looked at the documentation site and was not able to find the component.
- [ ] I have looked into the Figma boards and was not able to find the component.
- [x] I would be willing to provide a PR for this component.

# Story

As a user of the Synergy Design System, I would like to have a divider so that elements can be visually separated or grouped..

# Acceptance criteria

- [ ] All DoD items successfully completed
- [x] The component looks like the Figma design and reflects all its states
- [x] Incorporates slots, Events, Methods, Parts from Shoelace
- [x] Tested: Playwright + Chromatic
- [x] create a proper documentation which contains at least everything from the Shoelace component
- [ ] Text in the documentation can be synced

### PR DESCRIPTION
**Pull request by kirchsuSICKAG**
# Pull Request

## 📖 Description

This PR adds support for the divider component. It includes:

- The vendored component
- Storybook stories
- The generated angular, react and vue wrappers

### 🎫 Issues

Closes #270 

## ✅ DoD

- [x] I have tested my changes manually.
- [x] I have added automatic tests for my changes (unit- and visual regression tests).
- [x] I have updated the project documentation to reflect my changes (e.g. CHANGELOG, Storybook, ...).
- [x] I have made sure to follow the projects coding and contribution guides.
- [x] I have made sure that the bundle size has changed appropriately.
- [x] I have validated that there are no accessibility errors.
- [x] I have used design tokens instead of fix css values

### CODE REVIEW

**Comment by mariohamann** [comment]
There's quite a lot of stuff which doesn't belong here?

CleanShot 2024-01-18 at 14 38 11@2x

File: packages/docs/stories/components/divider.stories.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
+// ---------------------------------------------------------------------
+// 🔒 AUTOGENERATED BY VENDORISM
+// Removing this comment will prevent it from being managed by it.
+// ---------------------------------------------------------------------
+
+/* eslint-disable */
+
+/* eslint-disable import/no-relative-packages */
```
Comment (mariohamann) [inline packages/docs/stories/components/divider.stories.ts:8]: Hm. Should this be removed here? Something's a bit strange in the diff.

File: packages/docs/src/helpers/component.ts [hunk 2]
```diff
@@ -610,7 +611,7 @@ export const generateStoryDescription = <T extends Component>(component: T, attr
  * @param heightPx - the height of each single story
  * @returns the bundled story
  */
-export const generateScreenshotStory = (stories: Array<StoryObj>, heightPx: number = 150): StoryObj => {
+export const generateScreenshotStory = (stories: { [key: string]: StoryObj }, heightPx: number = 150): StoryObj => {
```
Comment (schilchSICKAG) [inline packages/docs/src/helpers/component.ts:614]: :) Muuuch better ;)

**Review by schilchSICKAG — APPROVED** [review]
Nice work

**Comment by kirchsuSICKAG** [comment]
> There's quite a lot of stuff which doesn't belong here?
> 
> CleanShot 2024-01-18 at 14 38 11@2x

Was resolved with a separate PR: https://github.com/synergy-design-system/synergy-design-system/pull/273

**Review by reichst1SICKAG — APPROVED** [review]
Design in chromatic looks good. Thanks

**Pull request merged**